### PR TITLE
Added "ESP" effect

### PR DESCRIPTION
**Pull request by OnlyRealNubs**
image
Colour of the boxes is randomized at the beginning of the effect. 
If you don't know what an ESP is: https://forums.daybreakgames.com/ps2/index.php?threads/what-does-esp-stand-for.160877/

### CODE REVIEW

**Comment by ReiDaTecnologia** [comment]
HAX!1!!

File: ChaosMod/Util/Graphics.h [hunk 1]
```diff
@@ -0,0 +1,20 @@
+
```
Comment (pongo1231) [inline ChaosMod/Util/Graphics.h:1]:
```suggestion
#pragma once
```

File: ChaosMod/Util/Graphics.h [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+#include "Natives.h"
+
+namespace Util
+{
+	inline void DRAW_2D_BOX(float x, float y, float width, float height, CRGBA colour, float lineThickness)
```
Comment (pongo1231) [inline ChaosMod/Util/Graphics.h:6]:
```suggestion
	inline void Draw2DBox(float x, float y, float width, float height, CRGBA colour, float lineThickness)
```
Also CRGBA struct should probably be moved here (and renamed to just RGBA)

File: ChaosMod/Util/Random.h [hunk 3]
```diff
@@ -28,6 +30,18 @@ class Random
...
+	inline CRGBA GetRandomColourRGB(unsigned int min = 0, unsigned int max = 255)
+	{
+		return { (unsigned int)GetRandomInt(min, max), (unsigned int)GetRandomInt(min, max), (unsigned int)GetRandomInt(min, max), max };
+	}
+
+	// Returns a random RGBA value with a random alpha.
+	inline CRGBA GetRandomColourRGBA(unsigned int min = 0, unsigned int max = 255)
+	{
+		return { (unsigned int)GetRandomInt(min, max), (unsigned int)GetRandomInt(min, max), (unsigned int)GetRandomInt(min, max), (unsigned int)GetRandomInt(min, max) };
+	}
```
Comment (pongo1231) [inline ChaosMod/Util/Random.h:44]: Those should not be here, move them to either Util or just keep them inside the effect

File: ChaosMod/stdafx.h [hunk 4]
```diff
@@ -2,43 +2,45 @@
```
Comment (pongo1231) [inline ChaosMod/stdafx.h:2]: Weird things going on here

File: ChaosMod/Effects/db/Misc/MiscEsp.cpp [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+		}
+	}
+}
+
+static RegisterEffect registerEffect(OnStart, nullptr, OnTick, EffectInfo
+	{
+		.Name = "ESP",
+		.Id = "misc_esp",
+		.IsTimed = true,
+		.IsShortDuration = true,
```
Comment (pongo1231) [inline ChaosMod/Effects/db/Misc/MiscEsp.cpp:44]: What's the reason for marking this one short timed?

**Review by pongo1231 — CHANGES_REQUESTED** [review]

File: ChaosMod/stdafx.h (hunk 4 above)
Comment (OnlyRealNubs) [reply ChaosMod/stdafx.h:2]: Believe this is just a merge conflict. I'll fix it rq

**Comment by OnlyRealNubs** [comment]
MoneyWasted made a better one so yuh

**Comment by ReiDaTecnologia** [comment]
I wonder if he knew this existed lol

**Comment by OnlyRealNubs** [comment]
He does

**Comment by ReiDaTecnologia** [comment]
"Fine. I will do it myself."

**Pull request closed without merging**